src/datasets.jl

```diff
@@ -351,12 +351,53 @@ function Base.setindex!(dset::Dataset, X::Array{T}, I::IndexType...) where {T}
     return X
 end
 
+function Base.setindex!(
+    dset::Dataset, X::Array{S}, I::IndexType...
+) where {S<:AbstractString}
```

Comment:
There is a fair amount of copying between this method and the above method.

The main difference is the need for `p = Ref{Cstring}(X)`. Can we abstract this somehow?